src/mongocxx/options/find_one_and_update.hpp

```diff
@@ -27,6 +26,17 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace options {
 
 ///
+/// Argument for find_one_and_update and find_one_and_replace to determine
+/// whether to return the document before or after modification.
+///
+/// @see https://docs.mongodb.org/manual/reference/method/db.collection.findAndModify/#return-data
+///
+enum class return_document {
```

Comment:
Thinking this should be in a separate place.  I don't know if the current file scheme allows for this, common things between options.

The Ruby driver had this concept of composable classes through mix-ins, which map loosely to multiple inheritance.  In that model we'd have some sort of "modify_return_order" option that both find_one_and_replace and find_one_and_update would include.  This might be a helpful model to handle other commonalities between options (for example implementing a "hintable" option class, and having all commands that take hints have "hintable" option classes)